src/Compilers/Core/Portable/PEWriter/MetadataWriter.DynamicAnalysis.cs

```diff
@@ -88,12 +88,17 @@ private BlobHandle GetOrAddBlob(ImmutableArray<byte> blob)
                 index = MetadataTokens.BlobHandle(_blobHeapSize);
                 _blobs.Add(blob, index);
 
-                _blobHeapSize += (blob.Length <= 0x7f) ? 1 : ((blob.Length <= 0x3fff) ? 2 : 4) + blob.Length;
```

Comment:
Missing parentheses. ```+``` has precedence over ```?:```. If the blob length was <= 0x7f it added just 1 byte to the size.